docs/specs/code/TRANSACTION_MEMO.md

```diff
@@ -0,0 +1,239 @@
+
+In terms of user interaction in standard Yoroi wallets, the user will continue to use the spending password to send transactions with memos without noticing any change, as the same password will decrypt the master key for create the transaction and decrypt the memo password used to encrypt memos. To read the memo of an existing transaction, the user will have to introduce his/her spending password.
+
+For hardware wallets, we propose to imitate the user-flow used in standard wallets. That is, the user will have to introduce his/her spending password at the time of sending a transaction **if and only if** he/she decides to add a memo. Similarly, the user will be asked for the spending password when reading a memo associated with a transaction.
+
+Regarding the signing process, notice that `ChaCha20Poly1305` includes a MAC which can be used to verify *integrity* and *authenticity* of the data. For the sake of simplicity, we consider that signing the memo to prove *ownership* is not necessary, as having a valid memo password implies having access to the spending password and the specific wallet (with its account public key).
+
+A Proof-of-Concept can be found [here](https://repl.it/repls/DimgreyImmaterialDebuggers).
+
+We use the Cardano encrypted message format in the same way as before. Similarly, the ciphertext is the result of applying 'password_ecnrypt' to the content of the memo.
+
+**Cardano encrypted message**
+```
+16(
+  [
+    / protected / h'' / {
+      \ is_ascii \ is_ascii:false \
+    } / ,
+    / unprotected / {
+      \ enc_type \ enc_type:1 \
+      \ version \ version:1 \
+    } /, 
+    / ciphertext / h'56294e41264b6eeaf9952f7e452ef05bdbeb8b3985a5ca05a88e848ce989483edb3783d92ba0c04cbdd4911825a3c0aebab032513ca4caea553d741434582969d356cd380587361b'
+  ]
+)
+```
+
+The above output is saved to a file named after the `Blake2b` hash of the corresponding transaction id and uploaded to the external storage. To decrypt we do the opposite process.
+
+### Public key vs User password
+
+We make a brief comparison of both methods advantages/disadvantages.
+
+| # | Description           | Public key | User password |
+ |-------|-----------------------|------------|-------|
+ | 1  | Stores seed/password encrypted | NO | YES |
+ | 2  | Signs the data to prove ownership | YES | NO |
+ | 3  | Verifies authenticity & integrity | YES | YES |
+ | 4  | Requires the user to enter a password to add/view memos | NO | YES |
+ | 4  | Adds extra steps to hardware wallets users | NO | YES |
+ | 5  | Available in Yoroi existing libraries | YES | YES |
+
+## External storage
+
+The proposed external storage is Dropbox. Based on its [Javascript SDK](https://github.com/dropbox/dropbox-sdk-js#readme), the API endpoints of our interest are:
+
+* [Create folder](https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesCreateFolder__anchor): Create a folder at a given path.
```

Comment:
Need a format for this path

Maybe

"cardano" / "mainnet"  / "transaction-memos"